Trends report

erfanio opened the pull request:
Trends report is available to collaborators at https://llm-exp.oss-fuzz.com/trend-reports/index.html

This PR
* Adds trends report web page
* Adds a cloud run function to update `trend-reports/index.json` in GCS
* Adds a cloud run function to update web page files (html, js, css)
* Adds a script to deploy cloud run functions (this script needs to be run manually after every change to the code)
* Moves trends report summary generation tool to a subdirectory so everything related to trends report is together

@DavidKorczynski @DonggeLiu please let me know if you want anything else added to the trends report

DonggeLiu commented:
Wow this is truly amazing, thanks @erfanio!
It makes comparing metrics across experiments so much easier, e.g., we can easily see the build rate improvements in 2024 Nov.

I just returned to work and might need a bit more time to read the code, but by looking at the trend report I think it's already great!

erfanio commented:
I've addressed most of [#issuecomment-2611279903](https://github.com/google/oss-fuzz-gen/pull/765#issuecomment-2611279903)

* I think it's best to fix (1) in a separate pull request. We need to change the docker_run script to either have a hook to upload incomplete data when it times out or continually upload the summary json which might cause other problems.
* For (5) I just added a legend 

https://llm-exp.oss-fuzz.com/trend-reports/index.html has the updated version PTAL

DonggeLiu commented:
Thanks! The new report looks perfect!

> I think it's best to fix (1) in a separate pull request. We need to change the docker_run script to either have a hook to upload incomplete data when it times out or continually upload the summary json which might cause other problems.

Yep! Let do it in a separate one then : )

DonggeLiu commented:
/gcbrun skip
